Source/WebKit/GPUProcess/media/cocoa/RemoteMediaPlayerProxyCocoa.mm

```diff
@@ -99,20 +103,33 @@
     m_layerHostingContextIDRequests.append(WTFMove(completionHandler));
 }
 
-void RemoteMediaPlayerProxy::setVideoLayerSizeFenced(const WebCore::FloatSize& size, WTF::MachSendRight&& machSendRight)
+void RemoteMediaPlayerProxy::setVideoLayerSizeFenced(const WebCore::FloatSize& size, WTF::MachSendRight&& machSendRight, CompletionHandler<void()>&& completionHandler)
 {
     ALWAYS_LOG(LOGIDENTIFIER, size.width(), "x", size.height());
-    if (m_inlineLayerHostingContext)
+
+#if USE(EXTENSIONKIT)
+    RetainPtr<_SEHostingUpdateCoordinator> hostingUpdateCoordinator;
+#endif
+
+    if (m_inlineLayerHostingContext) {
+#if USE(EXTENSIONKIT)
+        auto xpcRepresentation = adoptOSObject(xpc_dictionary_create(nullptr, nullptr, 0));
+        xpc_dictionary_set_mach_send(xpcRepresentation.get(), machPortKey, machSendRight.sendRight());
+        hostingUpdateCoordinator = adoptNS([alloc_SEHostingUpdateCoordinatorInstance() initFromXPCRepresentation:xpcRepresentation.get()]);
+        [hostingUpdateCoordinator addHostable:m_inlineLayerHostingContext->hostable().get()];
+#else
         m_inlineLayerHostingContext->setFencePort(machSendRight.sendRight());
+#endif
+    }
 
     m_configuration.videoLayerSize = size;
     setVideoLayerSizeIfPossible(size);
 
-    m_player->setVideoLayerSizeFenced(size, WTFMove(machSendRight));
+    m_player->setVideoLayerSizeFenced(size, WTFMove(machSendRight), [] { });
 
 #if USE(EXTENSIONKIT)
-    if (m_inlineLayerHostingContext)
-        m_inlineLayerHostingContext->commit();
+    [hostingUpdateCoordinator commit];
+    completionHandler();
```

Comment:
We only use `completionHandler` if `USE(EXTENSIONKIT)` is true. I think this might create a compile error for platforms that don't use `EXTENSIONKIT`; you might need to add a `UNUSED_PARAM(completionHandler)` as an `#else` here.

Reply:
Alternatively (and probably better style-wise) would be to call the completionHandler outside of the `#if USE(EXTENSIONKIT)` block.

Reply:
Looking below at `MediaPlayerPrivateRemote::setVideoLayerSizeFenced`, it seems even more wrong that we pass a no-op completion handler, then run code that invokes the completion handler, since `setVideoLayerSizeFenced` clearly triggers async remote calls in some cases. I think that means we will sometimes call our completion handler before the async work completes, which doesn't seem right.

Reply:
Yes, I believe you are right. I have addressed this and the compile issue in the latest version.